tidy3d/plugins/waveguide/rectangular_dielectric.py

```diff
@@ -210,41 +215,51 @@ def _set_array(cls, val):
             raise ValidationError("Values may not be negative.")
         return result
 
+    @pydantic.validator("clad_medium", always=True)
+    def _set_medium_list(cls, val):
+        """Normalize individual medium to single-element list."""
+        if isinstance(val, MediumType):
+            return [val]
+        return list(val)
+
     @pydantic.validator("box_medium", always=True)
     @skip_if_fields_missing(["clad_medium"])
     def _set_box_medium(cls, val, values):
         """Set BOX medium same as cladding as default value."""
-        return values["clad_medium"] if val is None else val
+        return values["clad_medium"][:1] if val is None else cls._set_medium_list(val)
 
     @pydantic.validator("clad_thickness", always=True)
     @skip_if_fields_missing(["clad_medium", "wavelength"])
     def _set_clad_thickness(cls, val, values):
         """Set default cladding thickness based on the max wavelength in the cladding medium."""
         if val is None:
             wavelength = values["wavelength"]
-            medium = values["clad_medium"]
+            medium = values["clad_medium"][0]
             n = numpy.array([medium.nk_model(f)[0] for f in C_0 / wavelength])
             lda = wavelength / n
-            return 1.5 * lda.max()
-        return val
+            return numpy.array([1.5 * lda.max()])
```

Comment:
consider setting 1.5 as a constant somewhere, or a field, if it would make sense to.